Code/GraphMol/Wrap/MolOps.cpp

```diff
@@ -305,9 +305,15 @@ void addRecursiveQueriesHelper(ROMol &mol, python::dict replDict,
   addRecursiveQueries(mol, replacements, propName);
 }
 
-ROMol *addHs(const ROMol &orig, bool explicitOnly = false,
-             bool addCoords = false) {
-  return MolOps::addHs(orig, explicitOnly, addCoords);
+ROMol *addHs(const ROMol &orig, bool explicitOnly, bool addCoords,
+             python::object onlyOnAtoms) {
+  std::vector<unsigned int> *onlyOn = NULL;
+  if (onlyOnAtoms) {
+    onlyOn = pythonObjectToVect(onlyOnAtoms, orig.getNumAtoms());
+  }
+  ROMol *res = MolOps::addHs(orig, explicitOnly, addCoords, onlyOn);
```

Comment:
if this throws an exception, isn't onlyOn leaked since it reaches delete?

(normally RAII takes care of this stuff, but I have no idea how boost::python deals with this).